### PR TITLE
Add the core of the fabulist ECS support.

### LINKED ISSUE #394: Integrate Fabulist into NovelRT
After #383 we're currently stuck without any form of interface for non-programmers. We should implement [Fabulist](https://github.com/NovelRT/Fabulist), our new VN language into NovelRT.

The idea is roughly:
- Fabulist is used to write VN contents
- Something else is used to store data for animations/scenes/etc (maybe Lua based too?)
- Something *else* is used to manage save states and such. (Fabulist will have a serializable "state" payload which represents the current state in the story, including things like variables)

### PR DESCRIPTION
**Pull request by RubyNova**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](https://github.com/novelrt/NovelRT/blob/misc/templates/Contributing.md#pull-requests)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This change adds the required ECS support for the Fabulist narrative scripting language. It allows you to execute narrative scrips via gameplay APIs.

**Is there an open issue that this resolves? If so, please [link it here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).**
Resolves #394 

**What is the current behavior?** (You can also link to an open issue here)
Fabulist does nothing.

**What is the new behavior (if this is a feature change)?**
Fabulist will be available by default as part of the gameplay APIs.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

**Other information**:
N/A

### CODE REVIEW

**Review by capnkenny — CHANGES_REQUESTED** [review]
Requesting changes for nits and also quite a few questions before we merge in.

Also, 2 things - 
1) I apologise in advance for commenting stuff out and not deleting 😅
2) We may need to consider that this will break the existing install logic - I propose that it's _not_ handled in this PR, and we just handle it as part of the future SDK work that will be starting back up soon.

File: CMakeLists.txt [hunk 1]
```diff
@@ -126,7 +126,7 @@ if(NOVELRT_FETCH_DEPENDENCIES)
       COMMAND_ERROR_IS_FATAL LAST
     )
     execute_process(
-      COMMAND ${CMAKE_COMMAND} --build ${CMAKE_CURRENT_BINARY_DIR}/depbuild --config Release -j${CORES_AMT}
+      COMMAND ${CMAKE_COMMAND} --build ${CMAKE_CURRENT_BINARY_DIR}/depbuild --config Release -j${CORES}
```
[the remark this answers is not in the document]
Comment (RubyNova) [reply CMakeLists.txt:129]: Is there no way we can use the -j value from the root invocation?

File: CMakeLists.txt [hunk 2]
```diff
@@ -165,11 +165,11 @@ if(NOVELRT_DEPS_INSTALLED)
   #So we just use our version that doesn't :)
   find_package(ZLIB CONFIG REQUIRED)
   include(${NOVELRT_DEPS_PATH}/lib/libpng/libpng16.cmake)
-  include(${NOVELRT_DEPS_PATH}/lib/fabulist.cmake)
+  #include(${NOVELRT_DEPS_PATH}/lib/fabulist.cmake)
```
[the remark this answers is not in the document]
Comment (RubyNova) [reply CMakeLists.txt:168]: Where is fabulist's build files located now?

File: include/NovelRT/Ecs/Audio/AudioEmitterComponent.h [hunk 3]
```diff
@@ -26,6 +26,17 @@ namespace NovelRT::Ecs::Audio
             volume = handleChanged ? other.volume : volume + other.volume;
             return *this;
         };
+
+        friend inline bool operator==(const AudioEmitterComponent& lhs, const AudioEmitterComponent& rhs) noexcept
```
[the remark this answers is not in the document]
Comment (RubyNova) [reply include/NovelRT/Ecs/Audio/AudioEmitterComponent.h:30]: It let us compile!

As part of a bug fix, an ECS tweak had to be made.

File: include/NovelRT/Ecs/Audio/AudioEmitterStateComponent.h [hunk 4]
```diff
@@ -23,6 +23,19 @@ namespace NovelRT::Ecs::Audio
             fadeExpectedVolume = other.fadeExpectedVolume;
             return *this;
         };
+
+        friend inline bool operator==(const AudioEmitterStateComponent& lhs,
```
[the remark this answers is not in the document]
Comment (RubyNova) [reply include/NovelRT/Ecs/Audio/AudioEmitterStateComponent.h:27]: CMake exiting with code 0. :D

File: src/NovelRT/Ecs/Narrative/NarrativePlayerSystem.cpp [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+                    if (_optionSelected)
+                    {
+                        if (_storyInstanceState->update())
+                        {
+                            _optionSelected = false;
+                            auto availableChoices = _catalogueForFrame->GetComponentView<ChoiceMetadataComponent>();
+                            for (auto&& [entity, choice] : availableChoices)
+                            {
+                                availableChoices.RemoveComponent(entity);
+                            }
```
[the remark this answers is not in the document]
Comment (RubyNova) [reply src/NovelRT/Ecs/Narrative/NarrativePlayerSystem.cpp:229]: This is my bad. Been tracking a lot this evening. I'll clean this up.

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (capnkenny) [reply CMakeLists.txt:129]: Nope, the root -j invocation happens _after_ this step, so we have no semblance without manually checking.

[the remark this answers is not in the document]
Comment (RubyNova) [reply CMakeLists.txt:129]: Added a new flag to account for this.

**Review by capnkenny — APPROVED** [review]
LGTM

**Pull request merged**